core/chains/evm/logpoller/expression.go

```diff
@@ -0,0 +1,167 @@
+package logpoller
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+)
+
+type qSelect struct {
+	Fields       []string
+	Table        string
+	WithCursor   bool
+	WithSequence bool
+}
+
+func (q qSelect) String() string {
+	selectAll := fmt.Sprintf("%s.*", q.Table)
+	fieldsOut := q.Fields
+
+	if len(q.Fields) == 0 {
+		fieldsOut = []string{selectAll}
+	}
+
+	if q.WithCursor {
+		fieldsOut = append(fieldsOut, fmt.Sprintf("%s AS %s", cursorFieldName, cursorFieldAlias))
+	}
+
+	// TODO: if sort by has sequence, add sequence to fields; still don't know composes a sequence but it's different from a cursor
+	if q.WithSequence {
+		fieldsOut = append(fieldsOut, fmt.Sprintf("id AS %s", sequenceFieldAlias))
```

Comment:
Since the sequence/cursor is a constructed field, the field name doesn't exist in the table. It must be constructed in the SELECT fields like `CONCAT(...) AS sequence` where `sequence` is the alias field name. In this case `id` is wrong and doesn't need to be aliased. `id` needs to be replaced with the `CONCAT` bit instead.

Reply:
It doesn't need to be a field, it just needs to convert to something like ORDER BY (block_number, log_index)`

Reply:
This is what log poller is doing right now, so I think we can do it here too